yarn-project/noir-contracts/src/contracts/child_contract/src/main.nr

```diff
@@ -5,44 +5,58 @@ contract Child {
     use dep::aztec::abi;
     use dep::aztec::abi::PrivateContextInputs;
     use dep::aztec::abi::PublicContextInputs;
-    use dep::aztec::context::Context;
+    use dep::aztec::context::{
+        PrivateContext,
+        PublicContext
+    };
     use crate::storage::Storage;
 
     fn constructor(
         inputs: PrivateContextInputs,
     ) -> distinct pub abi::PrivateCircuitPublicInputs {
         // Return private circuit public inputs. All private functions need to return this as it is part of the input of the private kernel.
-        Context::new(inputs, 0).finish()
+        PrivateContext::new(inputs, 0).finish()
     }
 
     // Returns a sum of the input and the chain id and version of the contract in private circuit public input's return_values.
     fn value(
         inputs: PrivateContextInputs,
         input: Field,
     ) -> distinct pub abi::PrivateCircuitPublicInputs {
-        let mut context = Context::new(inputs, abi::hash_args([input]));
+        let mut context = PrivateContext::new(inputs, abi::hash_args([input]));
 
-        context.return_values.push(input + inputs.private_global_variables.chain_id + inputs.private_global_variables.version);
+        context.return_values.push(input + context.chain_id() + context.version());
 
         // Return private circuit public inputs. All private functions need to return this as it is part of the input of the private kernel.
         context.finish()
     }
 
     // Returns base_value + 42.
-    open fn pubValue(inputs: PublicContextInputs, base_value: Field) -> pub Field {
-        base_value + inputs.public_global_variables.chain_id + inputs.public_global_variables.version + inputs.public_global_variables.block_number + inputs.public_global_variables.timestamp
+    open fn pubValue(inputs: PublicContextInputs, base_value: Field) -> pub abi::PublicCircuitPublicInputs {
+        let mut context = PublicContext::new(inputs, abi::hash_args([base_value]));
+        
+        // TODO: make these available on context
+        let returnValue = base_value + context.chain_id() + context.version() + context.block_number() + context.timestamp();
+
+        context.return_values.push(returnValue);
+        // TODO(MADDIAA): MAYBE we put the return values inside the finish object? That could have nice UX
```

Comment:
I think return_values are part of the PublicCircuitPublicInputs, although I think there are plans to instead return a `return_values_hash`. When that change happens, the simulator will keep track of the `return_values`, and the simulator could return those.